root/io/evolution/rules/execSourceTypes.cxx

```diff
@@ -237,13 +275,31 @@ bool check(T *p, V ref)
    return !error;
 }
 
+template<size_t N, typename V> //  = decltype(T::f)[N]>
+bool check_array(int arr[N], V ref)
```

Comment:
I guess that we could make the two parameters of the same type (esp. given that they are being compared 1-to1 below), right?

Reply:
Not quite.  `V` is meant to be any kind of collection (of int) with an `operator[]`